gulpfile.ts

```diff
@@ -0,0 +1,85 @@
+import { Octokit } from '@octokit/rest' // eslint-disable-line import/no-extraneous-dependencies
+import { series } from 'gulp' // eslint-disable-line import/no-extraneous-dependencies
+import { flatten } from 'lodash' // eslint-disable-line import/no-extraneous-dependencies
+import path from 'path'
+import { array, Codec, GetType, nullType, oneOf, string } from 'purify-ts/Codec'
+
+import { Decoder } from './src/helpers/decoder'
+
+const GithubObject = Codec.interface({
+  name: string,
+  path: string,
+  download_url: oneOf([string, nullType]),
+})
+type GithubObject = GetType<typeof GithubObject>
+
+interface APIModel extends GithubObject {
+  command: string
+}
+
+// Using authentication to increases Github API rate limit.
+const octokit = new Octokit()
+const owner = 'amzn'
+const repo = 'selling-partner-api-models'
+
+async function fetchContentsByPath(repoPath = 'models'): Promise<GithubObject[]> {
+  return octokit.repos
+    .getContent({
+      owner,
+      repo,
+      path: repoPath,
+    })
+    .then((response) =>
+      Array.isArray(response.data)
+        ? Decoder.decode(array(GithubObject), response.data)
+        : [Decoder.decode(GithubObject, response.data)],
+    )
+}
+
+function generateAPIModel(model: GithubObject): APIModel {
+  const outputFolder = path.basename(path.dirname(model.path))
+  const out = `src/api-models/${outputFolder}`
+  const command = `openapi-generator-cli generate -g typescript-axios --additional-properties=supportsES6=true,useSingleRequestParameter=true --type-mappings=set=Array --skip-validate-spec -o ${out} -i ${model.download_url}`
+
+  return {
+    ...model,
+    command,
+  }
+}
+
+function executeGeneratorCLI(model: APIModel): APIModel {
+  // TODO: generate model from API model
+
+  return model
+}
+
+function removeRedundantObjects(model: APIModel): APIModel {
+  /**
+   * TODO: clean up:
+   *- .openapi-generator
+   *- .gitignore
+   *- .openapi-generator-ignore
+   *- git_push.sh
+   */
+
+  return model
+}
+
+function exportAPIModel(model: APIModel): APIModel {
+  // TODO: export models into src/index.ts and commit files
+
+  return model
+}
+
+async function generateModels() {
+  const githubFolders = await fetchContentsByPath()
```

Comment:
When I `console.log` this, I get:

```js
[
  {
    name: 'aplus-content-api-model',
    path: 'models/aplus-content-api-model',
    download_url: null
  },
  {
    name: 'authorization-api-model',
    path: 'models/authorization-api-model',
    download_url: null
  },
  {
    name: 'catalog-items-api-model',
    path: 'models/catalog-items-api-model',
    download_url: null
  },
  {
    name: 'fba-inbound-eligibility-api-model',
    path: 'models/fba-inbound-eligibility-api-model',
    download_url: null
  },
  {
    name: 'fba-inventory-api-model',
    path: 'models/fba-inventory-api-model',
    download_url: null
  },
  {
    name: 'fba-small-and-light-api-model',
    path: 'models/fba-small-and-light-api-model',
    download_url: null
  },
  {
    name: 'feeds-api-model',
    path: 'models/feeds-api-model',
    download_url: null
  },
  {
    name: 'finances-api-model',
    path: 'models/finances-api-model',
    download_url: null
  },
  {
    name: 'fulfillment-inbound-api-model',
    path: 'models/fulfillment-inbound-api-model',
    download_url: null
  },
  {
    name: 'fulfillment-outbound-api-model',
    path: 'models/fulfillment-outbound-api-model',
    download_url: null
  },
  {
    name: 'merchant-fulfillment-api-model',
    path: 'models/merchant-fulfillment-api-model',
    download_url: null
  },
  {
    name: 'messaging-api-model',
    path: 'models/messaging-api-model',
    download_url: null
  },
  {
    name: 'notifications-api-model',
    path: 'models/notifications-api-model',
    download_url: null
  },
  {
    name: 'orders-api-model',
    path: 'models/orders-api-model',
    download_url: null
  },
  {
    name: 'product-fees-api-model',
    path: 'models/product-fees-api-model',
    download_url: null
  },
  {
    name: 'product-pricing-api-model',
    path: 'models/product-pricing-api-model',
    download_url: null
  },
  {
    name: 'reports-api-model',
    path: 'models/reports-api-model',
    download_url: null
  },
  {
    name: 'sales-api-model',
    path: 'models/sales-api-model',
    download_url: null
  },
  {
    name: 'sellers-api-model',
    path: 'models/sellers-api-model',
    download_url: null
  },
  {
    name: 'services-api-model',
    path: 'models/services-api-model',
    download_url: null
  },
  {
    name: 'shipping-api-model',
    path: 'models/shipping-api-model',
    download_url: null
  },
  {
    name: 'solicitations-api-model',
    path: 'models/solicitations-api-model',
    download_url: null
  },
  {
    name: 'uploads-api-model',
    path: 'models/uploads-api-model',
    download_url: null
  }
]
```

How do we use the `download_url` later? How does it work?

Reply:
@moltar : `download_url` will be available when it is file type. Your logs are `dir`. Therefore it will be `null`.